### ISSUE TITLE
Limited options and not all of them are being really used

### ISSUE THREAD

**Issue by eRadical** [mrlesmithjr/ansible-mariadb-galera-cluster#29]
Hi,

Acording to the defaults file

https://github.com/mrlesmithjr/ansible-mariadb-galera-cluster/blob/9622a6d7e5a1b673556c48a3a801d067847678ee/defaults/main.yml#L105-L127

you can make some changes but unfortunately there is no one size fits all and some of them are not being used in 

https://github.com/mrlesmithjr/ansible-mariadb-galera-cluster/blob/9622a6d7e5a1b673556c48a3a801d067847678ee/templates/etc/mysql/my.cnf.j2#L11-L30

@mrlesmithjr do you have any plans/thoughts on enhancing this?
[For example: I have bigger transaction than the default ib_logfile of 48MB ]

**Comment by mrlesmithjr** [comment]
@eRadical I did have plans on enhancing these but never did get around to them obviously ;)

**Comment by eRadical** [comment]
Can you share the plans?... I might give it a try... although it's very weird to tackle this:

- on Debian the service is started - and thus it is creating ibdata & ib_logfile*
- on CentOS - the rpm does mysql_install_db thus creating ibdata & ib_logfile*

We might get along not changing ibdata and only changing ib_logfile* by just removing them (stop server, remove files, start server).

Again what were your thoughts?

**Comment by eRadical** [comment]
I was thinking about something in the line of:

```
mariadb_extra_settings:
  mysql
    collation_server: 'latin1_swedish_ci' 
    character_set_client: 'latin1' 
    datadir: /var/lib/mysql 
    expire_logs_days: 10 
    ....................................
  mysqldump
    some_mysql_dump_var: value
    ....................................
  group:
    some_var: some_value

```

And the j2 file changed accordingly...

**Comment by mrlesmithjr** [comment]
Honestly I had nothing specific in mind. Just knew it was something that I would like to do at some point but never got around to it. However your ideas are more than welcomed dude!

**Comment by eRadical** [comment]
Another option and it's backward compatible:
- I added the variable mariadb_config_overrides with the same proposed structure as the above mariadb_extra_settings
- If the variable is defined it adds a new file /etc/my.cnf.d/overrides.cnf 

The problem w/ this approach is that I need to inject this in tasks/redhat.yml and in tasks/debian.yml just before the package is installed so the rpm finds the file and does not overwrite it... this solves the ib_logfile* issue and also the ib_data issue... BUT has another issue: the folder structure is not there yet and we need to create them first.

What's your take on this?

**Comment by mrlesmithjr** [comment]
I do not see any issue pre-creating the folder structure required to accomplish this if it makes sense.

**Comment by eRadical** [comment]
Fixed by #32
